src/scanspec/core.py

```diff
@@ -36,24 +46,15 @@
 ]
 
 
-class StrictConfig(BaseConfig):
-    """Pydantic configuration for scanspecs and regions."""
-
-    extra: Extra = Extra.forbid
+StrictConfig: ConfigDict = {"extra": "forbid"}
 
 
-def discriminated_union_of_subclasses(
-    super_cls: Optional[Type] = None,
-    *,
-    discriminator: str = "type",
-    config: Optional[Type[BaseConfig]] = None,
-) -> Union[Type, Callable[[Type], Type]]:
+def discriminated_union_of_subclasses(cls):
```

Comment:
Let's leave discriminator as a possible override if possible, it just needs to be passed into _TaggedUnion's __init__ and then used on line 168